contract/src/main.rs

```diff
@@ -731,7 +775,10 @@ pub extern "C" fn burn() {
 
     let token_identifier = utils::get_token_identifier_from_runtime_args(&identifier_mode);
 
-    let caller = utils::get_verified_caller().unwrap_or_revert();
+    let caller: Key = match utils::get_verified_caller().unwrap_or_revert() {
```

Comment:
get_verified_caller is sending one or two values in an enum so you need to match against them, afaik it would be hard here to make deduced casting, we could put the match into the method but this would complicate the caller handling return.